### ISSUE TITLE
Clarification on block wrapping closure body 

### ISSUE THREAD

**Issue by calebcartwright** [rust-dev-tools/fmt-rfcs#157]
We had a question come up in rustfmt around formatting of a closure, and I just wanted to clarify/confirm my reading of the style guide.

Specifically, my reading of the guide is that a closure like the one below should have the body block wrapped, and want to confirm this is correct:

```rust
let x = ||
    // a comment
    some_body_expr;
```

Should be formatted as:
```rust
let x = || {
    // a comment
    some_body_expr
};
```

That's based on the relevant section of [the guide for closures](https://github.com/rust-dev-tools/fmt-rfcs/blob/master/guide/expressions.md#closures) (emphasis mine)

> Use closures without the enclosing {}, if possible. Add the {} when you have a return type, when there are statements, **there are comments in the body**, or the body expression spans multiple lines and is a control-flow expression. 

Technically the comment is not within the span associated with the body, but the comment in this case still looks/feels like it meets the "comments in the body" criteria.

**Comment by joshtriplett** [comment]
On Fri, Aug 28, 2020 at 06:56:34AM -0700, Caleb Cartwright wrote:
> We had a question come up in rustfmt around formatting of a closure, and I just wanted to clarify/confirm my reading of the style guide.
> 
> Specifically, my reading of the guide is that a closure like the one below should have the body block wrapped, and want to confirm this is correct:
> 
> ```rust
> let x = ||
>     // a comment
>     some_body_expr;
> ```
> 
> Should be formatted as:
> ```rust
> let x = || {
>     // a comment
>     some_body_expr
> };
> ```
> 
> That's based on the relevant section of [the guide for closures](https://github.com/rust-dev-tools/fmt-rfcs/blob/master/guide/expressions.md#closures) (emphasis mine)
> 
> > Use closures without the enclosing {}, if possible. Add the {} when you have a return type, when there are statements, **there are comments in the body**, or the body expression spans multiple lines and is a control-flow expression. 
> 
> Technically the comment is not within the span associated with the body, but the comment in this case still looks/feels like it meets the "comments in the body" criteria.

This was the case we had in mind when we wrote that, yes. The above
formatting looks correct.

Perhaps it'd be clearer to say "when there are comments within the
closure".

(Also, for clarity and consistency, each of those cases should start
with "when".)

**Comment by calebcartwright** [comment]
> This was the case we had in mind when we wrote that, yes. The above
formatting looks correct.

Excellent thanks!

> Perhaps it'd be clearer to say "when there are comments within the
closure".
>
> (Also, for clarity and consistency, each of those cases should start
with "when".)

Agreed. I'll open a PR to tweak the text in the guide accordingly

**Comment by calebcartwright** [comment]
Closed via #158
